doc/introduction/logging.rst

```diff
@@ -0,0 +1,22 @@
+Logging
+=======
+
+PennyLane has support for Python's logging framework. PennyLane defines all logging configuration in a static TOML file, ``log_config.toml``, which can be customized to support rules for a given user or system environment.
+
+To see the default logging options you can explore the contexts of ``log_config.toml`` at the path given from :func:`pennylane.logging.config_path`.
```

Comment:
Is it intentional that this function doesn't have docstrings?

Reply:
It should have docstrings. At least, they are defined in the src.

Reply:
Do you expect the docstrings to be visible here, or somewhere else?

Reply:
I mean in the [docs](https://xanaduai-pennylane--4383.com.readthedocs.build/en/4383/introduction/logging.html), you can't click through to read about these functions.

Reply:
I think the ask here is to just add a `qml_logging.rst` page to `/doc/code`

Reply:
Ah, I understand now. Yep, @timmysilv is right, I'll add those in quickly.

Reply:
Updated to include the links, and added some additional references to the dev guidelines